File: git-epic-creator/services/code_graph_ingestion_service/src/core/records.py
"""Internal record contracts for graph persistence."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Optional


Language = str  # "cobol" | "java" | "javascript" | "xml" | "other"


@dataclass(frozen=True)
class CodeNodeRecord:
    project_id: str
    repo_fingerprint: str
    node_id: str
    language: Language
    kind: str
    symbol: Optional[str]
    file_path: str
    start_line: int
    end_line: int
    snippet_hash: str
    text: str
    extra_labels: tuple[str, ...] = ()


@dataclass(frozen=True)
class EdgeRecord:
    project_id: str
    repo_fingerprint: str
    rel_type: str
    src_node_id: str
    dst_node_id: str
    confidence: float
    metadata: dict

File: git-epic-creator/services/code_graph_ingestion_service/src/core/records.rs
use serde::{Deserialize, Serialize};
use serde_json::{Map, Value};

use crate::core::types::{CodeLanguage, CodeRelType};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CodeNodeRecord {
    pub project_id: String,
    pub repo_fingerprint: String,
    pub node_id: String,
    pub language: CodeLanguage,
    pub kind: String,
    pub symbol: Option<String>,
    pub file_path: String,
    pub start_line: i64,
    pub end_line: i64,
    pub snippet_hash: String,
    pub text: String,
    #[serde(default)]
    pub extra_labels: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EdgeRecord {
    pub project_id: String,
    pub repo_fingerprint: String,
    pub rel_type: CodeRelType,
    pub src_node_id: String,
    pub dst_node_id: String,
    pub confidence: f64,
    #[serde(default)]
    pub metadata: Map<String, Value>,
}


